code/datums/wounds/burns.dm

```diff
@@ -0,0 +1,334 @@
+		return
+	if(infestation <= 0 || infestation < sanitization)
+		to_chat(user, "<span class='notice'>There's no infection to treat on [victim]'s [limb.name]!</span>")
+		return
+
+	user.visible_message("<span class='notice'>[user] flashes the burns on [victim]'s [limb] with [I].</span>", "<span class='notice'>You flash the burns on [user == victim ? "your" : "[victim]'s"] [limb.name] with [I].</span>", vision_distance=COMBAT_MESSAGE_RANGE)
+	sanitization += I.uv_power
+	I.uv_cooldown = world.time + I.uv_cooldown_length
+
+/datum/wound/burn/proc/shave(obj/item/I, mob/user)
+	if(mortification <= 0)
+		to_chat(user, "<span class='notice'>There's no dead skin to shave from [victim]'s [limb.name]!</span>")
+		return
+
+	user.visible_message("<span class='danger'>[user] begins crudely carving away dead flesh from [victim]'s [limb.name] with [I].</span>", "<span class='notice'>You begin crudely carving dead flesh from [victim]'s [limb.name] with [I]...</span>", ignored_mobs=victim, vision_distance=COMBAT_MESSAGE_RANGE)
+	to_chat(victim, "<span class='danger'><b>[user] begins crudely carving away dead flesh from your [limb.name] with [I]!</b></span>") // und ZATS how I lost my medical license!
+	if(!do_after(user, base_treat_time * I.toolspeed, target=victim, extra_checks = CALLBACK(src, .proc/still_exists))) // TODO: be mean and make this a chance to fail, maybe increased success if they're drunk or unconscious
+		return
+
+	user.visible_message("<span class='danger'>[user] carves away some of the dead flesh from [victim]'s [limb.name]!</span>", "<span class='notice'>You carve away some of the dead flesh from [victim]'s [limb.name]!</span>", ignored_mobs=victim, vision_distance=COMBAT_MESSAGE_RANGE)
+	to_chat(victim, "<span class='danger'><b>[user] crudely carves away some of the dead flesh from your [limb.name]!</b></span>")
+	limb.receive_damage(brute=I.force, wound_bonus=CANT_WOUND) // i'll be very nice and not wound for now!
+	victim.bleed(I.force) // not TOO nice though
+	mortification = max(0, mortification - (I.force * 0.05)) // 12 force scalpel removes .6 mortification
+	if(prob(60))
+		victim.emote("scream")
+
+	if(mortification > 0)
+		try_treating(I, user)
+	else
+		to_chat(user, "<span class='green'>You've hacked away enough dead flesh from [victim]'s [limb.name] to let it recover.</span>")
+		to_chat(victim, "<span class='green'>[user] finishes hacking enough dead flesh from your [limb.name] to let it recover.</span>")
+
+/datum/wound/burn/treat(obj/item/I, mob/user)
+	if(istype(I, /obj/item/stack/medical/gauze))
+		bandage(I, user)
+	else if(istype(I, /obj/item/stack/medical/ointment))
+		ointment(I, user)
+	else if(istype(I, /obj/item/stack/medical/mesh))
+		mesh(I, user)
+	else if(istype(I, /obj/item/flashlight/pen/paramedic))
+		uv(I, user)
+	else if(I.sharpness)
+		shave(I, user)
+
+/datum/wound/burn/proc/regenerate_flesh(amount)
+	flesh_healing += amount * 0.5 // 20u patch will heal 10 flesh standard
```

Comment:
why are you halving it HERE instead of just halving the amount put into this proc on synthflesh's end?